### PR TITLE
Sam/put repo whitelisting

### PR DESCRIPTION
**Pull request by samholmes**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -31,6 +32,19 @@ export const asEdgeBox = asObject({
 export type StoreFileTimestampMap = ReturnType<typeof asStoreFileTimestampMap>
 export const asStoreFileTimestampMap = asMap(asNumber)
 
+// Settings Document
+
+export type StoreSettings = ReturnType<typeof asStoreSettings>
+export type StoreSettingsDocument = ReturnType<typeof asStoreSettingsDocument>
+export const asStoreSettings = asObject({
+  ipWhitelist: asArray(asString),
```
Comment (paullinator) [inline src/types.ts:40]: The white list and API key list should be maps instead of arrays

File: src/whitelisting.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const clientApiKey = req.query.apiKey
+
+  if (apiKeyWhitelist.length > 0 && !apiKeyWhitelist.includes(clientApiKey)) {
+    next(makeApiClientError(403, 'Forbidden API Key'))
+  }
+
+  next()
+}
+
+export const whitelistAll = Router()
```
Comment (paullinator) [inline src/whitelisting.ts:39]: will express return a failure if only one of the conditions succeeds? The server should allow either a valid API key or a valid IP address. I just want to make sure that we don’t fail the call if one passes and the other does not

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 1 above)
Comment (samholmes) [reply src/types.ts:40]: I'll change to `asMap(asBoolean)`.

File: src/whitelisting.ts (hunk 2 above)
Comment (samholmes) [reply src/whitelisting.ts:39]: If the condition is an OR, then I'll change this router composition.

**Review by paullinator — APPROVED** [review]
In the future, re-base the changes that fix issues versus creating new commits

**Pull request merged**